### DISCUSSION TITLE
Welcome to laravel-astronomy-library Discussions!

### DISCUSSION THREAD

**Discussion by WimDeMeester** [DeepskyLog/laravel-astronomy-library#49] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by SonjaNube** [comment]
Hello Wim, my name is Sonja and I stumbled upon this great librabry. I am trying to get it running but unfortunately I get stuck. The installation is finished, the database tables are updated, but using the code in my controller gives me the error: "Class "App\Http\Controllers\GeographicalCoordinates" not found". Did I miss part of the installation? Hope it is okay to pist here :)

**Reply by WimDeMeester** [reply]
Hi Sonja,

I guess you need to import GeographicalCoordinates first:

`use deepskylog\AstronomyLibrary\Coordinates\GeographicalCoordinates;`

Can you test that and report back if it works?

**Reply by SonjaNube** [reply]
Thanks for the fast reply! Oke, so use deepskylog\AstronomyLibrary was not enough, my fault.

use deepskylog\AstronomyLibrary\Coordinates\GeographicalCoordinates; works for my last error

but Class "App\Http\Controllers\AstronomyLibrary" not found is not fixed by 
use deepskylog\AstronomyLibrary\Coordinates\AstronomyLibrary, but by use deepskylog\AstronomyLibrary\AstronomyLibrary

how do I know which files I need to include?

my next error is: microsecond cannot be changed to float value -33244.161170306, integer expected
using the provided example:
$coords = new GeographicalCoordinates(-70.73330, -29.25);
$astrolib = new AstronomyLibrary($carbonDate, $coords);

**Reply by SonjaNube** [reply]
How do I know what to fill in at coords? is this a fixed given?
My goal with this program is to be able to get the coords of certain planets on a certain date and time :)

**Reply by WimDeMeester** [reply]
You should give it the geographical coordinates, so the coordinates of your observing location on earth.

**Reply by SonjaNube** [reply]
$coords_Utrecht = ['latitude' => 52.0919255, 'longitude' => 5.1229572];
$carbonDate = Carbon::now();

$coords = new GeographicalCoordinates($coords_Utrecht ['latitude'], $coords_Utrecht ['longitude']);
$astrolib = new AstronomyLibrary($carbonDate, $coords);

Error: microsecond cannot be changed to float value 638140.38487707, integer expected

$carbonDate = Carbon::createFromFormat('Y-m-d', $i['dateofbirth']);
Error: microsecond cannot be changed to float value -273403.10190549, integer expected

**Reply by WimDeMeester** [reply]
Hi @SonjaNube,
There was indeed a problem when using Carbon::now().  I always constructed my date myself, and I never used milliseconds, which Carbon::now() does.  
I have created a new version of the library (6.2.1).  Can you update to the newest version and re-test?

**Reply by SonjaNube** [reply]
can I email you? I get a lot of errors with composer update, but I have some other things going wrong to, so no clue what is from your tool, and not keen to post all here ;)

**Reply by WimDeMeester** [reply]
Yes, you can!  Can you send me your email-address?
